### PR TITLE
Use CalendarContract.Instances to fetch recurring events

### PR DESCRIPTION
**Pull request by umazalakain**
This addresses #158 by using `CalendarContract.Instances` to search for (possibly recursive events). It orders events by start time. It also simplifies querying: before we had `1+n` queries for `n` calendars, it is a single query now.

### CODE REVIEW

**Comment by billthefarmer** [comment]
Thank you for this.

**Comment by umazalakain** [comment]
@billthefarmer Thank you for merging it!

Actually, I realised there is a small mistake that will probably break insertion, I will add a comment pinpointing it.

File: src/main/java/org/billthefarmer/diary/QueryHandler.java [hunk 1]
```diff
@@ -107,8 +105,8 @@ public static void insertEvent(Context context, long startTime,
             queryHandler = new QueryHandler(resolver);
 
         ContentValues values = new ContentValues();
-        values.put(Events.DTSTART, startTime);
-        values.put(Events.DTEND, endTime);
+        values.put(Instances.BEGIN, startTime);
+        values.put(Instances.END, endTime);
```
Comment (umazalakain) [inline src/main/java/org/billthefarmer/diary/QueryHandler.java:109]: These should have stayed `Events.DTSTART` and `Events.DTEND`.

Comment (umazalakain) [reply src/main/java/org/billthefarmer/diary/QueryHandler.java:109]: @billthefarmer

**Comment by billthefarmer** [comment]
OK, got that. I hadn't got around to testing it.

**Comment by umazalakain** [comment]
I tested the event fetching part but not the insertion, supposedly I wasn't modifying that 🙃

**Comment by billthefarmer** [comment]
OK, tested and pushed to GitHub. I'm not going to post a new version until F-Droid have published the two versions in the pipeline. They're built, but not out there yet.

**Comment by umazalakain** [comment]
Awesome! I had no idea Fdroid was this slow. I found programming with android-studio really easy though, going from programming to building and installing the debug apk on the phone takes less than half a minute, I was pleasantly surprised.

**Comment by billthefarmer** [comment]
Well it varies wildly, sometimes an update get published the next day. I think they sometimes have problems with their bot driven automatic runs. Anyway, they have published new versions of about five of my apps in one hit. So I have pushed this update.

**Pull request merged**